test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/scaling/SmartRouterUpdateRedirection.java

```diff
@@ -0,0 +1,127 @@
+                .withMavenRepoUrl(MavenConstants.getMavenRepoUrl())
+                .withMavenRepoUser(MavenConstants.getMavenRepoUser(), MavenConstants.getMavenRepoPassword())
+                .withKieServerSyncDeploy(true)
+                .withApplicationName("kie-server-1")
+                .withSmartRouterConnection(smartRouterHostname, smartRouterPort)
+                .build();
+        DeploymentSettings kieServer2 = deploymentScenarioFactory.getKieServerS2ISettingsBuilder()
+                .withMavenRepoUrl(MavenConstants.getMavenRepoUrl())
+                .withMavenRepoUser(MavenConstants.getMavenRepoUser(), MavenConstants.getMavenRepoPassword())
+                .withKieServerSyncDeploy(true)
+                .withApplicationName("kie-server-2")
+                .withSmartRouterConnection(smartRouterHostname, smartRouterPort)
+                .build();
+        DeploymentSettings smartRouter = deploymentScenarioFactory.getSmartRouterSettingsBuilder()
+                .withHostame(smartRouterHostname)
+                .build();
+
+        return deploymentScenarioFactory.getGenericScenarioBuilder()
+                .withKieServer(kieServer1)
+                .withKieServer(kieServer2)
+                .withSmartRouter(smartRouter)
+                .build();
+    }
+
+    @Before
+    public void prepateClientsAndProject() {
+        KieServerDeployment kieServerDeployment1 = deploymentScenario.getKieServerDeployments().get(0);
+        KieServerDeployment kieServerDeployment2 = deploymentScenario.getKieServerDeployments().get(1);
+        SmartRouterDeployment smartRouterDeployment = deploymentScenario.getSmartRouterDeployments().get(0);
+
+        MavenDeployer.buildAndDeployMavenProject(ClassLoader.class.getResource("/kjars-sources/definition-project-snapshot").getFile());
+        MavenDeployer.buildAndDeployMavenProject(ClassLoader.class.getResource("/kjars-sources/definition-project-101-snapshot").getFile());
+
+        kieServerClient1 = KieServerClientProvider.getKieServerClient(kieServerDeployment1);
+        kieServerClient2 = KieServerClientProvider.getKieServerClient(kieServerDeployment2);
+        smartRouterClient = KieServerClientProvider.getSmartRouterClient(smartRouterDeployment,
+                kieServerDeployment1.getUsername(), kieServerDeployment1.getPassword());
+    }
+
+    @Test
+    public void testRouterLoadBalancing() {
+        deployProject(kieServerClient1, CONTAINER_ID, Project.DEFINITION_SNAPSHOT);
+        deployProject(kieServerClient2, CONTAINER_ID, Project.DEFINITION_SNAPSHOT);
+
+        verifyProcessAvailableInContainer(smartRouterClient, CONTAINER_ID, USERTASK_PROCESS_ID);
+
+        deployProject(kieServerClient1, CONTAINER_ID_UPDATED, Project.DEFINITION_101_SNAPSHOT);
+
+        for (int i = 0; i < RETRIES_NUMBER; i++) {
+            verifyProcessAvailableInContainer(smartRouterClient, CONTAINER_ID_UPDATED, UPDATED_USERTASK_PROCESS_ID);
```

Comment:
We should also verify this for CONTAINER_ALIAS.

Reply:
Yes, I plan to implement it as second test